brian2/codegen/optimisation.py

```diff
@@ -248,6 +248,16 @@ def render_BinOp(self, node):
                 # only simplify this if the type wouldn't be cast by the operation
                 if dtype_hierarchy[right.dtype] <= dtype_hierarchy[left.dtype]:
                     return left
+        elif op.__class__.__name__ == 'FloorDiv':
+            if left.__class__.__name__ == 'Num' and left.n == 0:  # 0//x
+                if node.stateless:
+                    # Do not remove stateful functions
+                    return _replace_with_zero(left, node)
+            # Only optimise floor division by 1 if the number is an integer, for
+            # floating point values, floor division by 1 changes the value
+            if (left.dtype == 'integer' and
+                    right.__class__.__name__ == 'Num' and right.n == 1):  # x//1
+                    return left
```

Comment:
I think this needs an extra check, because ``3//1.0`` should be ``3.0`` not ``3`` and I think this last bit will replace it with ``3``?

Reply:
Good catch!